src/util/config.rs

```diff
@@ -762,12 +763,19 @@ mod check_data_dir {
     }
 
     fn get_fs_info(path: &str, mnt_file: &str) -> Result<FsInfo, ConfigError> {
+        lazy_static! {
+            // According `man 3 getmntent`, The pointer returned by `getmntent` points
+            // to a static area of memory which is overwritten by subsequent calls.
+            // So we use a lock to protect it in order to avoid `make dev` fail.
+            static ref GETMNTENT_LOCK: Mutex<()> = Mutex::new(());
+        }
         unsafe {
             let profile = CString::new(mnt_file).unwrap();
             let retype = CString::new("r").unwrap();
             let afile = libc::setmntent(profile.as_ptr(), retype.as_ptr());
             let mut fs = FsInfo::default();
             loop {
+                let _ = GETMNTENT_LOCK.lock().unwrap();
```

Comment:
Looks like the lock guard is dropped here, so it unlocks immediately. And could you move `lock()` up to the `loop`?